ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/util/CommandParser.java

```diff
@@ -420,106 +352,22 @@ private static SqlAssertTopicCommand getAssertTopicCommand(
-    return checkAssert(tokens, TOPIC);
-  }
-
-  private static boolean isAssertSchema(final List<String> tokens) {
-    return checkAssert(tokens, SCHEMA);
-  }
-
-  private static boolean checkAssert(final List<String> tokens, final String assertType) {
-    if (tokens.size() >= 1 && tokens.get(0).equals(ASSERT)) {
-      final boolean notExists =
-          tokens.size() >= 4 && tokens.get(1).equals(NOT) && tokens.get(2).equals(EXISTS);
-      if (notExists) {
-        return tokens.get(3).equals(assertType);
-      } else {
-        return tokens.get(1).equals(assertType);
-      }
-    }
-    return false;
-  }
-
-  private static boolean isInsertValuesStatement(final List<String> tokens) {
-    // CHECKSTYLE_RULES.OFF: BooleanExpressionComplexity
-    return tokens.size() > 3
-        && tokens.get(0).equals(INSERT)
-        && tokens.get(1).equals(INTO)
-        && !tokens.get(3).equals(SELECT)
-        && tokens.contains(VALUES);
-    // CHECKSTYLE_RULES.ON: BooleanExpressionComplexity
-  }
-
-
-  private static boolean isCreateConnectorStatement(final List<String> tokens) {
-    // CHECKSTYLE_RULES.OFF: BooleanExpressionComplexity
-    return tokens.size() > 2
-        && tokens.get(0).equals(CREATE)
-        && (tokens.get(1).equals(SINK) || tokens.get(1).equals(SOURCE))
-        && tokens.get(2).equals(CONNECTOR);
-    // CHECKSTYLE_RULES.ON: BooleanExpressionComplexity
-  }
-
-  private static boolean isDropConnectorStatement(final List<String> tokens) {
-    return tokens.size() > 1 && tokens.get(0).equals(DROP) && tokens.get(1).equals(CONNECTOR);
-  }
-
-  private static boolean isSetStatement(final List<String> tokens) {
-    return tokens.size() > 0 && tokens.get(0).equals(SET);
-  }
-
-  private static boolean isUnsetStatement(final List<String> tokens) {
-    return tokens.size() > 0 && tokens.get(0).equals(UNSET);
-  }
-
-  private static boolean isDefineStatement(final List<String> tokens) {
-    return tokens.size() > 0 && tokens.get(0).equals(DEFINE);
-  }
-
-  private static boolean isUndefineStatement(final List<String> tokens) {
-    return tokens.size() > 0 && tokens.get(0).equals(UNDEFINE);
-  }
 
   /**
    * Validates that the sql statement represented by the list of input tokens
    * (keywords separated whitespace, or strings identified by single quotes)
    * is not unsupported by the migrations tool. Assumes that tokens have already
    * been upper-cased.
    *
-   * @param tokens components that make up the sql statement. Each component is
+   * @param sql components that make up the sql statement. Each component is
    *               either a keyword separated by whitespace or a string enclosed
    *               in single quotes. Assumes that tokens have already been upper-cased.
    */
-  private static void validateSupportedStatementType(final List<String> tokens) {
+  private static void validateSupportedStatementType(final String sql) {
```

Comment:
We can eliminate this. We can just do a check after parsing to see if the parsed statement type is one of the supported types.